python/kernmlops/cli/collect.py

```diff
@@ -52,19 +62,22 @@ def run_collect(
     system_info = system_info.unnest(system_info.columns)
     collection_id = system_info["collection_id"][0]
     output_dir = generic_config.get_output_dir() / "curated" if bpf_programs else generic_config.get_output_dir() / "baseline"
+    queue = Queue(maxsize=1)
 
     for bpf_program in bpf_programs:
         bpf_program.load(collection_id)
         if verbose:
             print(f"{bpf_program.name()} BPF program loaded")
     if verbose:
         print("Finished loading BPF programs")
+    poll_thread = Thread(target = poll_instrumentation, args = (benchmark, bpf_programs, queue, generic_config.poll_rate))
+    poll_thread.start()
```

Comment:
Thread should probably join() when complete

Reply:
I'm not sure it is required in the scope of our tool, but fair point.